<element name="world" required="*">
  <description>The world element encapsulates an entire world description including: models, scene, physics, joints, and plugins</description>

  <attribute name="name" type="string" default="__default__" required="1">
    <description>Unique name of the world</description>
  </attribute>

  <element name="max_step_size" type="double" default="0.001" required="1">
    <description>Maximum time step size at which every system in simulation can interact with the states of the world.  (was physics.sdf's dt).</description>
  </element>

  <!-- real_time_factor (simulation speedup) might be more intuitive to end
       users than real_time_update_rate -->
  <element name="real_time_factor" type="double" default="1.0" required="1">
    <description>target simulation speedup factor, defined by ratio of simulation time to real-time.</description>
  </element>

  <!-- to be deprecated by real_time_factor
  <element name="real_time_update_rate" type="double" default="1000" required="1">
    <description>Rate at which to update the physics engine (UpdatePhysics calls per real-time second). (was physics.sdf's update_rate).</description>
  </element>
  -->

  <include filename="gui.sdf" required="0"/>
  <include filename="physics.sdf" required="1"/>
  <include filename="scene.sdf" required="1"/>
  <include filename="light.sdf" required="*"/>

  <include filename="model.sdf" required="*"/>
  <include filename="actor.sdf" required="*"/>
  <include filename="plugin.sdf" required="*"/>
  <include filename="joint.sdf" required="*"/>
  <include filename="road.sdf" required="*"/>

  <include filename="state.sdf" required="*"/>
</element> <!-- End World -->
